x11-edge-grab/src/main.rs

```diff
@@ -32,58 +41,68 @@ enum State {
+    let config = Config::load(cfg_file).unwrap();
+    let control_socket = xdg_dir.place_runtime_file("control.sock").unwrap();
+    let mut writer = UnixStream::connect(control_socket).unwrap();
     let mut reader = writer.try_clone().unwrap();
     let (con, _) = Connection::connect(None).unwrap();
     let root = con.get_setup().roots().take(1).next().unwrap().root();
     register(&con, root);
     let state = Arc::new(Mutex::new(State::Detached));
 
     let state_r = state.clone();
+    let cfg = config.clone();
     thread::spawn(move || {
         loop {
             match unwrap!(reader.read_u8()) {
                 1 => {
-                    let x = unwrap!(reader.read_i32::<LittleEndian>());
-                    let y = unwrap!(reader.read_i32::<LittleEndian>());
+                    let xwin = unwrap!(reader.read_i32::<LittleEndian>());
+                    let ywin = unwrap!(reader.read_i32::<LittleEndian>());
                     if *unwrap!(state_r.lock()) != State::Attached {
                         continue;
                     }
-                    println!("{}:{}", x, y);
-                    if x <= 0 && y >= 540 && y < 1080+540
-                        || x >= 3839 && y >= 540 && y < 1080+540 {
-                        println!("exit");
-                        let x = x + 1920;
+                    // for now we only support one windows monitor
+                    let windows_monitor = cfg.monitors.iter().find(|m| m.is_windows).unwrap();
+                    // transform to linux coordinates
+                    let right = xwin >= windows_monitor.bounds.width - 1;
+                    let down = ywin >= windows_monitor.bounds.height - 1;
+                    let xlin = if right && !windows_monitor.connected {
+                        xwin + windows_monitor.bounds.x - windows_monitor.bounds.width
+                    } else {
+                        xwin + windows_monitor.bounds.x
+                    };
+                    let ylin = if down && !windows_monitor.connected {
+                        ywin + windows_monitor.bounds.y - windows_monitor.bounds.height
+                    } else {
+                        ywin + windows_monitor.bounds.y
+                    };
+                    if cfg.monitors.iter().any(|m| m.bounds.contains((xlin, ylin)) && !m.is_windows) {
                         unwrap!(reader.write_u8(4));
-                        println!("written");
                         unwrap!(reader.flush());
                         *unwrap!(state_r.lock()) = State::TryDetach;
-                        println!("flushed");
                         let mut cmd = Command::new("xdotool");
                         cmd.arg("mousemove")
-                            .arg(x.to_string())
-                            .arg(y.to_string());
+                            .arg(xlin.to_string())
+                            .arg(ylin.to_string());
```

Comment:
I feel like implementing this xdotool functionality wouldn't be too hard since you're already using xcb.